### PR TITLE
Extract RemoteAddress.Unknown when cannot find client ip

### PR DESCRIPTION
**Pull request by dyk**

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @dyk,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by dyk** [comment]
I did sign the CLA

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
Thanks, good idea - did we have a ticket for this?

**Comment by dyk** [comment]
> Thanks, good idea - did we have a ticket for this?

Nope

> Test FAILed.

Fixed

File: akka-docs/rst/scala/http/routing-dsl/directives/misc-directives/extractClientIP.rst [hunk 1]
```diff
@@ -11,7 +11,7 @@ Signature
 
 Description
 -----------
-Provides the value of ``X-Forwarded-For``, ``Remote-Address``, or ``X-Real-IP`` headers as an instance of ``HttpIp``.
+Provides the value of ``X-Forwarded-For``, ``Remote-Address``, or ``X-Real-IP`` headers as an instance of ``RemoteAddress``.
```
Comment (ktoso) [inline akka-docs/rst/scala/http/routing-dsl/directives/misc-directives/extractClientIP.rst]: thanks

**Comment by ktoso** [comment]
Resolves https://github.com/akka/akka/issues/21301

**Comment by ktoso** [comment]
> > Thanks, good idea - did we have a ticket for this?
> Nope

It should ;-) I added one :)

Thanks for the PR.

File: akka-http/src/main/scala/akka/http/scaladsl/server/directives/MiscDirectives.scala [hunk 2]
```diff
@@ -104,7 +104,8 @@ object MiscDirectives extends MiscDirectives {
   private val _extractClientIP: Directive1[RemoteAddress] =
     headerValuePF { case `X-Forwarded-For`(Seq(address, _*)) ⇒ address } |
       headerValuePF { case `Remote-Address`(address) ⇒ address } |
-      headerValuePF { case `X-Real-Ip`(address) ⇒ address }
+      headerValuePF { case `X-Real-Ip`(address) ⇒ address } |
+      extract[RemoteAddress] { case _ ⇒ RemoteAddress.Unknown }
```
Comment (2beaucoup) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/MiscDirectives.scala]: `provide(RemoteAddress.Unknown)` ?

Comment (ktoso) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/MiscDirectives.scala]: hah, good catch, you want to submit that as change @dyk or @2beaucoup ?

Comment (2beaucoup) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/MiscDirectives.scala]: on it...

Comment (ktoso) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/MiscDirectives.scala]: <3

**Pull request merged**